galite-core/src/main/kotlin/org/kopi/galite/form/VFixnumField.kt

```diff
@@ -18,12 +18,809 @@
+          throw VFieldException(this, MessageCode.getMessage("VIS-00010"))
+        }
+      }
+      return v
+    }
+  }
+
+  /**
+   * Returns the display representation of field value of given record.
+   */
+  override fun getTextImpl(r: Int): String {
+    if (value[r] == null) {
+      return ""
+    }
+    var res: String = toText(value[r]!!.setScale(currentScale[r]))
+
+    // append spaces until the max scale is reached to make commas aligned.
+    // append an extra space to replace the missing comma if the current scale is zero.
+    if (block.isMulti()) {
+      for (i in (if (currentScale[r] == 0) -1 else currentScale[r]) until fieldMaxScale) {
+        res += " "
+      }
+    }
+    return res
+  }
+
+  /**
+   * Returns the SQL representation of field value of given record.
+   */
+  override fun getSqlImpl(r: Int): String {
+    return if (value[r] == null) "NULL" else value[r]!!.toSql()
+  }
+
+  /**
+   * Copies the value of a record to another
+   */
+  override fun copyRecord(f: Int, t: Int) {
+    val oldValue = value[t]
+    value[t] = value[f]
+    // inform that value has changed for non backup records
+    // only when the value has really changed.
+    if ((t < block.bufferSize
+                    && (((oldValue != null && value[t] == null)
+                    || (oldValue == null && value[t] != null)
+                    || (oldValue != null && oldValue != value[t]))))) {
+      fireValueChanged(t)
+    }
+  }
+  /*
+   * ----------------------------------------------------------------------
+   * FORMATTING VALUES WRT FIELD TYPE
+   * ----------------------------------------------------------------------
+   */
+  /**
+   * Returns a string representation of a bigdecimal value wrt the field type.
+   */
+  fun formatFixed(value: org.kopi.galite.type.Fixed): String {
+    return toText(value.setScale(currentScale[block.activeRecord]))
+  }
+
+  /**
+   * Returns the string represention in human-readable format.
+   */
+  fun toText(v: Fixed): String {
+    return if (!fraction) {
+      v.toString()
+    } else {
+      toFraction(v.toString())
+    }
+  }
+
+  private fun toFraction(str: String): String {
+    var dot: Int
+    if ((str.indexOf(',').also { dot = it }) == -1) {
+      return str
+    }
+    val precomma = str.substring(0, dot)
+    val fract = Integer.valueOf(str.substring(dot + 1, str.length)).toInt()
+    when {
+      fract * 64 % 1000000 != 0 -> {
+        return str
+      }
+      fract == 0 -> {
+        return precomma
+      }
+      else -> {
+        var num: Int
+        var den = 64
+        num = (fract * den) / 1000000
+        while (num % 2 == 0) {
+          num /= 2
+          den /= 2
+        }
+        when {
+          precomma == "0" -> {
+            return "$num/$den"
+          }
+          precomma == "-0" -> {
+            return "-$num/$den"
+          }
+          else -> {
+            return "$precomma $num/$den"
+          }
+        }
```

Comment:
Refactor to : 
return when (precomma) {
          "0" -> "$num/$den"
          "-0" -> "-$num/$den"
          else -> "$precomma $num/$den"
        }